tools/check-links

```diff
@@ -105,8 +105,8 @@ class Crawler:
 
     def get_urls_to_crawl(self, packet) -> list:
         urls_from_page = list()
-        if packet.url[-1] == '/':
-            packet.url = urljoin(packet.url, 'index.html')
+        if packet.url[-1] == "/":
+            packet.url = urljoin(packet.url, "index.html")
```

Comment:
Almost-blocking: This puzzled me at first glance.

* We're changing `packet.url`, but the URL has already been used to download the page. Therefore, we won't actually request `/foo/index.html`. But the code makes you think that we'd request `/foo/index.html`.
* `url` and `defraged_url` are out of sync after this mutation.

If the goal of these lines is to influence line 110's logic, why not change the `in_allowed_file_soup` function instead? Is there some non-local side effect of this code that I'm not seeing?